File: Day-4/Alternate_array/Alternate_array.py
def alternate_array(arr):
    if len(arr) < 2:
        return arr

    result = []
    left, right = 0, len(arr) - 1

    while left <= right:
        if left == right:
            result.append(arr[left])
        else:
            result.append(arr[left])
            result.append(arr[right])
        left += 1
        right -= 1

    return result
if __name__ == "__main__":
    arr = [1, 2, 3, 4, 5]
    print(alternate_array(arr))  # Output: [1, 5, 2, 4, 3]
# Time Complexity: O(n) - We traverse the array once to create the alternate array.
# Space Complexity: O(n) - We create a new array to store the alternate elements.

File: Day-4/Alternate_array/Alternate_array.c
#include<sdio.h>
void alternate_array(int arr[],int n)
{
    int i,temp;
    for(i=0;i<n-1;i+=2)
    {
        temp=arr[i];
        arr[i]=arr[i+1];
        arr[i+1]=temp;
    }
}
int main()
{
    int n,i;
    printf("Enter the size of the array: ");
    scanf("%d",&n);
    int arr[n];
    printf("Enter the elements of the array: ");
    for(i=0;i<n;i++)
    {
        scanf("%d",&arr[i]);
    }
    alternate_array(arr,n);
    printf("Alternate array: ");
    for(i=0;i<n;i++)
    {
        printf("%d ",arr[i]);
    }
    return 0;
}
// Time Complexity: O(n)
// Space Complexity: O(1)
